### PR TITLE
[TTAHUB-2609/2610] Rearrange iPD course selection in activity report and session report forms

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
- On the course selector (session report and activity report forms use the same component), add "Other" as option for iPD course name.
- Sort order for iPD course names in alphabetical order except for "Other". "Other" is at the end of the course name list.
- Move iPD radio button above "Link to TTA resource" radio buttons.
- Change "Link to TTA resource" to "Did you use any other TTA resources that are available as a link ?"
- Change "Did you use any TTA resources that aren't available as a link ?" to "Did you use any other TTA resources that are available as a link ?"
- Selection is cached

## How to test
- Confirm the above is true and that you can save/edit/view objective courses in both forms
- Confirm that the design and behavior match the updated design for both forms

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2609
* https://jira.acf.gov/browse/TTAHUB-2610

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by kryswisnaskas — COMMENTED** [review]
Looks good! 👍 

The only question I had was on the font size. In the following it appears that the first "Did you use any other TTA..." has a smaller font than the second.

image

**Review by kryswisnaskas — APPROVED** [review]
Nice!

File: package.json [hunk 1]
```diff
@@ -151,7 +151,8 @@
       "node_modules/*",
       "playwright.config.js",
       "tests/*",
-      "packages/*"
+      "packages/*",
+      "frontend/public/*"
```
Comment (AdamAdHocTeam) [inline package.json:155]: should we have front/public in the BE package.json?

File: src/app.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ app.use((req, res, next) => {
       scriptSrc: ["'self'", '*.googletagmanager.com'],
       scriptSrcElem: ["'self'", 'https://*.googletagmanager.com', `'nonce-${res.locals.nonce}'`],
       imgSrc: ["'self'", 'data:', 'www.googletagmanager.com', '*.google-analytics.com'],
-      connectSrc: ["'self'", '*.google-analytics.com', '*.analytics.google.com', '*.googletagmanager.com'],
+      connectSrc: ["'self'", '*.google-analytics.com', '*.analytics.google.com', '*.googletagmanager.com', 'https://kit.fontawesome.com', 'https://ka-p.fontawesome.com'],
```
Comment (AdamAdHocTeam) [inline src/app.js:71]: Is this to remove a warning/error in the console?

File: src/migrations/20240506144701-update-course-model.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+      // add persistsOnUpload to the Courses table
+      await queryInterface.addColumn('Courses', 'persistsOnUpload', {
+        type: Sequelize.BOOLEAN,
+        allowNull: false,
+        defaultValue: false,
+      }, { transaction });
+
+      // insert an "other" course
+      await queryInterface.bulkInsert('Courses', [
```
Comment (AdamAdHocTeam) [inline src/migrations/20240506144701-update-course-model.js:18]: Not that it matters but should we be using bulk insert for a single entry?

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good. Some minor comments/questions.

File: src/migrations/20240506144701-update-course-model.js (hunk 3 above)
Comment (thewatermethod) [reply src/migrations/20240506144701-update-course-model.js:18]: yeah its the easiest way to do an insert via the ORM in a migration. evaluates to the same thing.

File: src/app.js (hunk 2 above)
Comment (thewatermethod) [reply src/app.js:71]: good catch, we actually don't need these, I will remove

File: package.json (hunk 1 above)
Comment (thewatermethod) [reply package.json:155]: yeah I think it needs to be there so that it doesn't try and lint that folder if we run lint from here on the CI or something

**Pull request merged**